src/cli/cli.js

```diff
@@ -1,16 +1,26 @@
 import _ from 'lodash';
-import { pkg } from '../utils';
+import { pkg, NodeVersion } from '../utils';
 import Command from './command';
 import serveCommand from './serve/serve';
 
 const argv = process.env.kbnWorkerArgv ? JSON.parse(process.env.kbnWorkerArgv) : process.argv.slice();
 const program = new Command('bin/kibana');
 
+// Validates current the NodeJS version compatibility when Kibana starts.
+NodeVersion.runValidator(() => {
```

Comment:
i'm thinking we'll want to run this as part of the server start process [here](https://github.com/elastic/kibana/blob/master/src/server/kbn_server.js#L75).

1) we'll have the standard logger available
1) we'll have config available

there's going to be more bootstrap checks in the future and we'll have more info available at that point.  the downside is we may fatal beforehand if someone's a breaking change, but i'm okay with that.  thoughts?

Reply:
Good point @jbudz. Thinking about the reasons (1 and 2)  you provide I really agree with you!

Reply:
I'd actually like to see this check at the first possible moment (before even babel-register) because requiring a specific node version sometimes means that we won't even be able to get to the server start process without hitting a syntax error or something.

Reply:
if i'm honest with myself this was a ploy to provide a config for disabling the version check.  in the above case
1) syntax error, server doesn't start
1) api has changed, server doesn't fatal, version check exits

as long as it's before any writes the worst case is a vague error, but the person knows they've been messing with things at that point.

i'm still not a fan of forcing an exact version match with no workarounds.  package mangers, manual security updates and swapping distributions all seem valid to me.  with that said i'm in the minority on this so lets leave it and if there's an issue we can revisit

Reply:
If the version check is in a module by itself we can offer a workaround 😉

```
echo '' > src/babel-register/node_version_check.js
```

Reply:
Please don't tell anyone to change Kibana source files :P

Reply:
whoops, by leave it as is i mean ignore my comment.  i missed the module part above.

Reply:
We almost went there in https://github.com/elastic/kibana/pull/19104/files/bd0f4e1bfe41abe39f9d61caa8a583b10cc80f58#diff-202f54e3b2f5233afda1e3c964211e5d, but maybe this pushes it over the edge and it's time to create a `src/setup_node_env.js` module or something that combines babel-register, ts-node/register, and the node version check...